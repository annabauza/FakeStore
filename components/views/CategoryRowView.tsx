import React from 'react';
import {View, Text, Button, StyleSheet} from 'react-native';
import {NavigationScreenProp} from 'react-navigation';

export interface CategoryRowViewProps {
  navigation: NavigationScreenProp<any, any>;
  category: string;
}

const CategoryRowView: React.FC<CategoryRowViewProps> = ({
  category,
  navigation,
}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.input}>{category}</Text>
      <Button
        title="more >"
        onPress={() => {
          navigation.navigate('Products', {categoryId: category});
        }}
      />
      <View style={styles.divider} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 16,
  },
  divider: {
    flex: 1,
    height: 1,
    backgroundColor: '#000',
    marginBottom: 12,
  },
  thumbnail: {
    flex: 1,
    width: 66,
    height: 66,
    resizeMode: 'contain',
  },
  logo: {
    width: 66,
    height: 58,
  },
  input: {
    margin: 16,
  },
});

export default CategoryRowView;
