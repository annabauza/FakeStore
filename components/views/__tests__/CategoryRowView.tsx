import React from 'react';
import renderer from 'react-test-renderer';
import CategoryRowView from '../CategoryRowView';

test('renders correctly', () => {
  const tree = renderer
    .create(<CategoryRowView navigation={undefined} category={''} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
