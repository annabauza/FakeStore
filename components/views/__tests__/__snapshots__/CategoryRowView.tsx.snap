// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 16,
    }
  }
>
  <Text
    style={
      Object {
        "margin": 16,
      }
    }
  >
    
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        more &gt;
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#000",
        "flex": 1,
        "height": 1,
        "marginBottom": 12,
      }
    }
  />
</View>
`;
